sdk/typescript/src/index.guard.ts

```diff
@@ -726,3 +734,172 @@ export function isTransactionResponse(obj: any, _argumentName?: string): obj is
             isPublishResponse(obj.PublishResponse) as boolean)
     )
 }
+
+export function isTransferObjectTx(obj: any, _argumentName?: string): obj is TransferObjectTx {
+    return (
+        (obj !== null &&
+            typeof obj === "object" ||
+            typeof obj === "function") &&
+        (obj.TransferObject !== null &&
+            typeof obj.TransferObject === "object" ||
+            typeof obj.TransferObject === "function") &&
+        isTransactionDigest(obj.TransferObject.recipient) as boolean &&
+        isSuiObjectRef(obj.TransferObject.object_ref) as boolean
+    )
+}
+
+export function isPublishTx(obj: any, _argumentName?: string): obj is PublishTx {
+    return (
+        (obj !== null &&
+            typeof obj === "object" ||
+            typeof obj === "function") &&
+        (obj.Publish !== null &&
+            typeof obj.Publish === "object" ||
+            typeof obj.Publish === "function") &&
+        (obj.Publish.modules !== null &&
+            typeof obj.Publish.modules === "object" ||
+            typeof obj.Publish.modules === "function") &&
+        typeof obj.Publish.modules["__@iterator"] === "function"
```

Comment:
I'm curious where `__@iterator` is coming from, seems like an internal property on something?